components/enrichers/reachability/internal/search/search.go

```diff
@@ -0,0 +1,108 @@
+package search
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/ocurity/dracon/components/enrichers/reachability/internal/atom"
+)
+
+// Searcher is responsible for finding reachable purls in the supplied reachability report.
+type Searcher struct {
+	reachablePurls  atom.ReachablePurls
+	reachables      atom.Reachables
+	matcherFileLine *regexp.Regexp
+}
+
+// NewSearcher returns a new searcher.
+func NewSearcher(reachables atom.Reachables, reachablePurls atom.ReachablePurls) (*Searcher, error) {
+	switch {
+	case len(reachables) == 0:
+		return nil, errors.New("invalid empty reachables")
+	case len(reachablePurls) == 0:
+		return nil, errors.New("invalid empty reachable purls")
+	}
+
+	matcherFileLine, err := regexp.Compile(`(?P<file>[^/]+):(?P<line>[\d-]+)`)
```

Comment:
does atom only have lines? 
i thought it had ranges, happy to be wrong if your tests show otherwise :)

Reply:
From my tests and my understanding it looks ok but please feel free to test too!